File: CarePackage/Software/Brave.cs
namespace CarePackage.Software;

public class Brave : BaseSoftware
{
    public override string           Key            { get; set; } = "brave-browser";
    public override string           Name           { get; set; } = "Brave";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Browser;
    public override string           DownloadName   { get; set; } = "BraveBrowserSetup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = true;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.brave_browser;
    public override string           Homepage       { get; set; } = "https://brave.com";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://laptop-updates.brave.com/download/desktop/release/BRV010?bitness=64");
}

File: packages/app/src/services/software/definitions/brave.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'brave-browser';
	public name = 'Brave';
	public category = [SoftwareCategory.Browser];
	public downloadName = 'BraveBrowserSetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'brave-browser.png';
	public homepage = 'https://brave.net';

	public async resolveDownloadUrl() {
		return ok('https://laptop-updates.brave.com/download/desktop/release/BRV010?bitness=64');
	}
}
